### PR TITLE
[Graphbolt] Dispatch edge ids in neighbor sampling

### PR DESCRIPTION
**Pull request by peizhou001**
## Description
This PR use torch built-in dispatch macro to dispatch types of edge ids, and change all related data access from 'item()' to raw pointer.
This PR combined with #5890 will improve the performance by around 30%.
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: graphbolt/src/csc_sampling_graph.cc [hunk 1]
```diff
@@ -141,37 +141,43 @@ c10::intrusive_ptr<SampledSubgraph> CSCSamplingGraph::SampleNeighbors(
...
+      indptr_.scalar_type(), "parallel_for", ([&] {
+        torch::parallel_for(0, num_nodes, 32, [&](scalar_t b, scalar_t e) {
+          const scalar_t* indptr_data = indptr_.data_ptr<scalar_t>();
+          for (scalar_t i = b; i < e; ++i) {
+            const auto nid = nodes[i].item<int64_t>();
+            TORCH_CHECK(
+                nid >= 0 && nid < NumNodes(),
+                "The seed nodes' IDs should fall within the range of the "
+                "graph's "
+                "node IDs.");
```
Comment (frozenbugs) [inline graphbolt/src/csc_sampling_graph.cc:154]:
```suggestion
                "graph's node IDs.");
```

**Comment by frozenbugs** [comment]
Can you also include the performance diff in this PR description?

File: graphbolt/src/csc_sampling_graph.cc (hunk 1 above)
Comment (peizhou001) [reply graphbolt/src/csc_sampling_graph.cc:154]: fixed

**Comment by peizhou001** [comment]
> Can you also include the performance diff in this PR description?

The number of tests performed has been added, and the detailed benchmark can only be run once the code has been merged into the main branch.

**Comment by frozenbugs** [comment]
@dgl-bot

**Review by frozenbugs — APPROVED** [review]

**Pull request merged**